.docker/Dockerfile

```diff
@@ -1,11 +1,10 @@
+# syntax=docker/dockerfile:1.7-labs
+# Above must be the first line in Dockerfile.  It enables the "copy --parents" syntax
```

Comment:
While we are talking about cleaning up the Dockerfile, something that I have been considering doing is removing the `ci` stage completely, leaving the `robot` stage as the initial stage. The `ci` stage doesn't particularly do much in this project that `ros-industrial/industrial_ci` can't do already. I've been doing some testing with this in a few different repos e.g., [`waterlinked_dvl` ci](https://github.com/Robotic-Decision-Making-Lab/waterlinked_dvl/blob/4411139f890271957610d35389a18dc409b539eb/.github/workflows/ci.yml#L23). The development packages like `clang-format-14` (which needs to get updated to `clang-format-18` sometime) and `clang-tidy` should be getting installed in the devcontainer Dockerfile anyway.

Reply:
Hm, the upstream_ / downstream_ options provide a lot of functionality.  I'm sure it could be bashed into building mavros/mavlink (potentially with some fork of the repo with the code patches installed).   You end up maintaining two separate build paths _and_ the fork of mavros.

From an efficiency point of view, the ci image is a "cached image where apt-get and rosdep have already been run" preventing a lot of redundant apt-getting.    I guess the question is whether that's worth it?

Reply:
What I would envision is pushing those installs to the `robot` stage. Though, with #241 still in the queue, I don't think this is something that we need to work on right now--more just something for us to think about for future discussion.